### PR TITLE
Add platform docs for FreeBSD.

### PR DESCRIPTION
**Pull request by asomers**

### CODE REVIEW

**Comment by asomers** [comment]
cc @tgross35 .  Also, perhaps @MikaelUrankar would like to sign up as a second target maintainer?

**Comment by ehuss** [comment]
r? ehuss

Looks like we have two PRs updating the same thing over at #128546.

Can we combine them? Or I can merge #128546 now, until this is ready?

**Review by workingjubilee — CHANGES_REQUESTED** [review]
In the platform support docs, please note that [FreeBSD's ports patches i686-unknown-freebsd](https://github.com/freebsd/freebsd-ports/blob/a35c8865a6883be6ecdcec286b7c2a620f69d3e6/lang/rust/files/patch-compiler_rustc__target_src_spec_targets_i686__unknown__freebsd.rs) in a way that [introduces incorrect compilations](https://github.com/rust-lang/rust/issues/114479), thus people must use the upstream toolchain instead of a FreeBSD-built toolchain for building for i686-unknown-freebsd without introducing incorrect compilations.

Unless this changes, of course.

**Comment by MikaelUrankar** [comment]
> cc @tgross35 . Also, perhaps @MikaelUrankar would like to sign up as a second target maintainer?

Yes, you can put me as a 2nd target maintainer.

**Comment by asomers** [comment]
@workingjubilee @beetrees the decision to build lang/rust without SSE support was made in 2017, before any of the linked LLVM issues.  I'll start a discussion about revisiting it.  I'll inquire about the ELFv1/ELFv2 issue too.

**Comment by workingjubilee** [comment]
> Ideally (since FreeBSD wants to support pre-SSE2 CPUs)

I am not entirely sure they do? I have been informed contrariwise, i.e. that FreeBSD is only interested in Pentium 4 or later equivalents. Naturally, _that_ might have been decided many years after the 2017 decision!

Of course, if two targets settles the matter for FreeBSD, that is fine by me, and if a documentation change is preferred, that is also fine by me. I do not mean to be obstructive, I merely want the matter to be clear for everyone involved.

File: src/doc/rustc/src/platform-support/freebsd.md
Comment (emaste) [inline src/doc/rustc/src/platform-support/freebsd.md:38]: maybe "the FreeBSD ports tree"

**Comment by ehuss** [comment]
Just FYI, you'll need to add the link to SUMMARY.md. Also, I'd suggest fixing the target definitions in 
`compiler/rustc_target/src/spec/targets` if those are indeed incorrect. See #128546 for examples of these changes.

**Comment by workingjubilee** [comment]
@ehuss I would rather not deliberately introduce miscompilations into existing tier 2 targets that currently have none, whether or not `ports` has decided to make their target unsound.

**Comment by ehuss** [comment]
@workingjubilee Sorry, I'm not following. Are you asking that the change [requested above](https://github.com/rust-lang/rust/pull/129220#pullrequestreview-2244183862) be resolved? I'm not the author, so I wouldn't be the one making that change. As the reviewer, I won't merge until it is resolved.

Or maybe you intended to tag one of the other commenters here? Sorry, I'm confused. 😕

**Comment by workingjubilee** [comment]
@ehuss I am saying that the resolution can include `ports` fixing its target definition to not make Rust code unsound.

**Comment by asomers** [comment]
A change is underway to reenable SSE on i386 by default: https://reviews.freebsd.org/D47227

**Comment by Dylan-DPC** [comment]
@asomers any updates on this? thanks

**Comment by asomers** [comment]
> @asomers any updates on this? thanks

I expect to merge that change, reenabling SSE2 on i386, but EoD on Saturday.

**Comment by tgross35** [comment]
Is the SSE note accurate since the push 3 weeks ago?

If not, I think it would be okay to drop that section for now and merge the rest of the docs so we at least have something.

**Comment by asomers** [comment]
> Is the SSE note accurate since the push 3 weeks ago?
> 
> If not, I think it would be okay to drop that section for now and merge the rest of the docs so we at least have something.

Yes, the SSE note is accurate.  I committed that change on 9-Nov.  https://reviews.freebsd.org/R11:b039f2e46b15c1ff5cbaf41e642993d982294da3

**Review by tgross35 — APPROVED** [review]
This needs an entry in SUMMARY.md and a squash, but with https://github.com/freebsd/freebsd-ports/commit/b039f2e46b15c1ff5cbaf41e642993d982294da3 I don't think there is anything that this would be waiting on.

**Comment by tgross35** [comment]
@asomers could you rebase and squash? I'll merge this, if any small changes are needed they can come as a followup.

**Comment by asomers** [comment]
> @asomers could you rebase and squash? I'll merge this, if any small changes are needed they can come as a followup.

done

**Pull request merged**
